connectors/src/connectors/microsoft/lib/graph_api.ts

```diff
@@ -82,6 +83,91 @@ export async function getFilesAndFolders(
   return { results: res.value };
 }
 
+export async function getDeltaResults({
+  client,
+  parentInternalId,
+  nextLink,
+  token,
+}: {
+  client: Client;
+  parentInternalId: string;
+} & (
+  | { nextLink: string; token?: never }
+  | { nextLink?: never; token: string }
+)) {
+  const { nodeType, itemAPIPath } = typeAndPathFromInternalId(parentInternalId);
+
+  if (nodeType !== "drive" && nodeType !== "folder") {
+    throw new Error(
+      `Invalid node type: ${nodeType} for getFilesAndFolders, expected drive or folder`
+    );
+  }
+
+  if (nextLink && token) {
+    throw new Error("nextLink and token cannot be used together");
+  }
+
+  const deltaPath =
+    (nodeType === "folder"
+      ? itemAPIPath + "/delta"
+      : itemAPIPath + "/root/delta") + (token ? `?token=${token}` : "");
+
+  const res = nextLink
+    ? await client.api(nextLink).get()
+    : await client
+        .api(deltaPath)
+        .header("Prefer", "odata.track-changes, deltaExcludeParent=true")
+        .get();
+
+  if ("@odata.nextLink" in res) {
+    return {
+      results: res.value,
+      nextLink: res["@odata.nextLink"],
+    };
+  }
```

Comment:
`nit`: Feel like this condition is used in a bunch of places, we could probably refactor it